getValue.php

```diff
@@ -50,13 +50,21 @@
 	} else {
 		$matchSource = (empty($matchSource)) ? $thismatch : $matchSource ;
 		$filterData = \REDCap::getData($_POST['otherpid'], 'array', null, null, null, null, false, false, false, "([$matchSource] = '$matchRecord')");
-		if(count($filterData) != 1){
-			// Either there were no matches or multiple matches.  Either way, we want to return without echo-ing any values.
-			return;
-		}
 
-		reset($filterData);
-		$recordId = key($filterData);
+		if (count($filterData) != 1) {
+			$useLatestRecordIdSetting = $module->getProjectSetting('use-latest-record-id');
+			if (!is_null($useLatestRecordIdSetting) && isset($useLatestRecordIdSetting[0]) && $useLatestRecordIdSetting[0] === true) {
+				$recordKeys = array_keys($filterData);
+				$records = \Records::getRecordList($_POST['otherpid'], [], false, false, null, null, 0, $recordKeys);
+				$recordId = end($records);
```

Comment:
The 'getRecordList' function could possibly return 'false' in some instances. It's not likely, but you may want some default behavior in case '$records' ends up being false. Something like $recordId = end($records ?: [array_key_first($filterData)]); If you use 'array_key_first' you'd have to set the min-php version in config.json to 7.3, because apparently some people are still using this module on PHP 5.

Reply:
Awesome thanks for bringing this to my attention. I don't want to force 7.3 if I don't need to, so I found and tested another solution.